h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/mapred/H2OYARNRunner.java

```diff
@@ -0,0 +1,80 @@
+package water.hadoop.mapred;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.ClientCache;
+import org.apache.hadoop.mapred.ResourceMgrDelegate;
+import org.apache.hadoop.mapred.YARNRunner;
+import org.apache.hadoop.mapreduce.MRJobConfig;
+import org.apache.hadoop.mapreduce.v2.app.H2OMRAppMaster;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.yarn.api.records.ApplicationSubmissionContext;
+import org.apache.hadoop.yarn.api.records.ContainerLaunchContext;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Vector;
+
+import water.util.CollectionUtils;
+
+/**
+ * A H2o specific ProtocolClient
+ *
+ * In this case we use standard YARNRunner, but just override application type to H2O.
+ */
+public class H2OYARNRunner extends YARNRunner {
+
+  private static final Log LOG = LogFactory.getLog(H2OYARNRunner.class);
+
+  private static final String APPLICATION_MASTER_CLASS = H2OMRAppMaster.class.getName();
+
+  public H2OYARNRunner(Configuration conf) {
+    super(conf);
+  }
+
+  public H2OYARNRunner(Configuration conf,
+                       ResourceMgrDelegate resMgrDelegate) {
+    super(conf, resMgrDelegate);
+  }
+
+  public H2OYARNRunner(Configuration conf, ResourceMgrDelegate resMgrDelegate,
+                       ClientCache clientCache) {
+    super(conf, resMgrDelegate, clientCache);
+  }
+
+  @Override
+  public ApplicationSubmissionContext createApplicationSubmissionContext(Configuration jobConf,
+                                                                         String jobSubmitDir,
+                                                                         Credentials ts)
+      throws IOException {
+    // Change created app context
+    LOG.info("Setting application type to H2O");
+    ApplicationSubmissionContext appContext = super.createApplicationSubmissionContext(jobConf, jobSubmitDir, ts);
+    appContext.setApplicationType("H2O");
+    // Modify MRAppMaster commands to point to our master
+    LOG.info("Setting MRAppMaster to " + H2OMRAppMaster.class.toString());
+    ContainerLaunchContext origClc = appContext.getAMContainerSpec();
+    ContainerLaunchContext newClc = ContainerLaunchContext.newInstance(
```

Comment:
If we only need to set application type to H2O then we don't need a custom application master yet.

Is this meant as a groundwork for the future improvements?